### PR TITLE
pep8 for schedules app

### PR DESCRIPTION
**Pull request by stefan-walluhn**
Make schedules app PEP8 conform. This changeset also includes some very minor code changes, that does not affect any program logic, but helps to improve readability. I also deleted some legacy commented code fragments that led to confusion while reading the code.

### CODE REVIEW

**Comment by txenoo** [comment]
I've just created an issue so Iago could link the pull request later to an issue.  #24

File: radio/apps/schedules/models.py [hunk 1]
```diff
@@ -227,45 +291,58 @@ def between(cls, after, before, exclude=None, live=False, schedule_board=None):
...
                 closer_date = date
                 closer_schedule = schedule
-        if closer_schedule is None:  # Todo: check
+        if closer_schedule is None:  # XXX Todo: check
             return None, None
         return closer_schedule, closer_date
 
     def __unicode__(self):
-        return self.get_day_display() + ' - ' + self.start_hour.strftime('%H:%M')
+        return ' - '.join(
```
Comment (stefan-walluhn) [inline radio/apps/schedules/models.py]: Whoops, while writing some unit tests it emerged that this is not working. We need a sequence. I'll provide a fix.

**Comment by stefan-walluhn** [comment]
fix committed

**Comment by iago1460** [comment]
@stefan-walluhn feel free to join the discussion at #24

File: radio/apps/schedules/admin.py [hunk 2]
```diff
@@ -78,9 +78,6 @@ def copy_ScheduleBoard(self, request, queryset):
     copy_ScheduleBoard.short_description = _("Make a Copy of calendar")
 
 
-admin.site.register(ScheduleBoard, ScheduleBoardAdmin)
-
-
```
Comment (stefan-walluhn) [inline radio/apps/schedules/admin.py]: bunch admin registration at the end of module to not spread registration logic across the code

File: radio/apps/schedules/admin.py [hunk 3]
```diff
@@ -97,8 +94,8 @@ def get_urls(self):
             'app_name': self.model._meta.app_label,
             'model_name': self.model._meta.model_name,
         }
-        custom_urls = patterns('',
-            url(
+        custom_urls = patterns(
+            '', url(
```
Comment (stefan-walluhn) [inline radio/apps/schedules/admin.py]: this was mangled by the pep8 tool (continuation line under-indented for visual indent)

File: radio/apps/schedules/admin.py [hunk 4]
```diff
@@ -111,11 +108,12 @@ def get_urls(self):
         return custom_urls + urls
 
     def response_change(self, request, obj):
-        msg = _('%(obj)s was changed successfully.') % {'obj': force_unicode(obj)}
+        msg = _('{obj} was changed successfully.'.format(obj=force_unicode(obj)))
```
Comment (stefan-walluhn) [inline radio/apps/schedules/admin.py]: I prefer format over "%" syntax, but this is just personal taste. Feel free to mangle this change.

File: radio/apps/schedules/admin.py [hunk 5]
```diff
@@ -111,11 +108,12 @@ def get_urls(self):
...
         if '_continue' in request.POST:
             return HttpResponseRedirect(request.path)
         else:
             return HttpResponseRedirect("../../")
 
 
+admin.site.register(ScheduleBoard, ScheduleBoardAdmin)
 admin.site.register(Schedule, FullcalendarAdmin)
```
Comment (stefan-walluhn) [inline radio/apps/schedules/admin.py]: bunch admin registration at the end of module

File: radio/apps/schedules/models.py [hunk 6]
```diff
@@ -61,11 +62,11 @@ def clean(self):
                 if self.start_date > self.end_date:
                     raise ValidationError(_('end date must be greater than or equal to start date.'))
                 # check date collision
-                qs = ScheduleBoard.objects.filter(
-                    start_date__lte=self.end_date, end_date__isnull=True
-                ) | ScheduleBoard.objects.filter(
-                    start_date__lte=self.end_date, end_date__gte=self.start_date
+                qs = (
+                    ScheduleBoard.objects.filter(start_date__lte=self.end_date, end_date__isnull=True) |
+                    ScheduleBoard.objects.filter(start_date__lte=self.end_date, end_date__gte=self.start_date)
```
Comment (stefan-walluhn) [inline radio/apps/schedules/models.py]: Make code more readable. The important part is to apply multiple filters, so the code should not spread this across multiple indentation levels, but should reflect this by intuitive reading.

File: radio/apps/schedules/models.py [hunk 7]
```diff
@@ -73,10 +74,9 @@ def clean(self):
 
             else:
                 # start_date != None and end_date == None only one can exist
-                qs = ScheduleBoard.objects.filter(
-                    start_date__isnull=False, end_date__isnull=True
-                ) | ScheduleBoard.objects.filter(
-                    end_date__gte=self.start_date
+                qs = (
+                    ScheduleBoard.objects.filter(start_date__isnull=False, end_date__isnull=True) |
+                    ScheduleBoard.objects.filter(end_date__gte=self.start_date)
```
Comment (stefan-walluhn) [inline radio/apps/schedules/models.py]: Make code more readable. The important part is to apply multiple filters, so the code should not spread this across multiple indentation levels, but should reflect this by intuitive reading.

File: radio/apps/schedules/models.py [hunk 8]
```diff
@@ -86,9 +86,6 @@ def clean(self):
 
         elif self.end_date:
             raise ValidationError(_('start date cannot be null if end date exists'))
-        else:
-            # both None = disable
-            pass
```
Comment (stefan-walluhn) [inline radio/apps/schedules/models.py]: This has no effect at all.

File: radio/apps/schedules/models.py [hunk 9]
```diff
@@ -104,11 +101,10 @@ def save(self, *args, **kwargs):
...
     def get_current(cls, dt):
-        schedule_board = cls.objects.filter(
-            start_date__lte=dt, end_date__isnull=True
-        ).order_by('-start_date') | cls.objects.filter(
-            start_date__lte=dt, end_date__gte=dt
-        ).order_by('-start_date')
+        schedule_board = (
+            cls.objects.filter(start_date__lte=dt, end_date__isnull=True).order_by('-start_date') |
+            cls.objects.filter(start_date__lte=dt, end_date__gte=dt).order_by('-start_date')
+        )
```
Comment (stefan-walluhn) [inline radio/apps/schedules/models.py]: Make code more readable. The important part is to apply multiple filters, so the code should not spread this across multiple indentation levels, but should reflect this by intuitive reading.

File: radio/apps/schedules/models.py [hunk 10]
```diff
@@ -149,24 +145,23 @@ def __get_rrule(self):
             # Due to rrule we need to add 1 day
             end_date = end_date + datetime.timedelta(days=1)
             return rrule.rrule(
-                rrule.WEEKLY, byweekday=[self.day], dtstart=datetime.datetime.combine(
-                    start_date, self.start_hour
-                ), until=end_date
-            )
+                rrule.WEEKLY, byweekday=[self.day],
+                dtstart=datetime.datetime.combine(start_date, self.start_hour),
+                until=end_date)
```
Comment (stefan-walluhn) [inline radio/apps/schedules/models.py]: Make code more readable, do not split datetime.combine function call across multiple lines resulting in spread rrule params where we are not forced to do so.

File: radio/apps/schedules/models.py [hunk 11]
```diff
@@ -149,24 +145,23 @@ def __get_rrule(self):
...
             if end_date:
                 # Due to rrule we need to add 1 day
                 end_date = end_date + datetime.timedelta(days=1)
                 return rrule.rrule(
-                    rrule.WEEKLY, byweekday=[self.day], dtstart=datetime.datetime.combine(
-                        start_date, self.start_hour
-                    ), until=end_date
+                    rrule.WEEKLY, byweekday=[self.day],
+                    dtstart=datetime.datetime.combine(start_date, self.start_hour),
+                    until=end_date
```
Comment (stefan-walluhn) [inline radio/apps/schedules/models.py]: Make code more readable, do not split datetime.combine function call across multiple lines resulting in spread rrule params where we are not forced to do so.

File: radio/apps/schedules/models.py [hunk 12]
```diff
@@ -149,24 +145,23 @@ def __get_rrule(self):
...
                 )
             else:
                 return rrule.rrule(
-                    rrule.WEEKLY, byweekday=[self.day], dtstart=datetime.datetime.combine(
-                        start_date, self.start_hour)
+                    rrule.WEEKLY, byweekday=[self.day],
+                    dtstart=datetime.datetime.combine(start_date, self.start_hour)
```
Comment (stefan-walluhn) [inline radio/apps/schedules/models.py]: Make code more readable, do not split datetime.combine function call across multiple lines resulting in spread rrule params where we are not forced to do so.

File: radio/apps/schedules/models.py [hunk 13]
```diff
@@ -209,16 +204,11 @@ def clean(self):
...
-                                        'start_month': start_date.strftime("%m"),
-                                        'start_year': start_date.strftime("%Y"),
-                                        'end_date': end_date.strftime("%H:%M"), 'end_day': end_date.strftime("%d"),
-                                        'end_month': end_date.strftime("%m"), 'end_year': end_date.strftime("%Y"),
-                                    }
+                                    _('This settings collides with: {name} [{start_date} - {end_date}]').format(
+                                        name=schedule.programme.name,
+                                        start_date=start_date.strftime("%H:%M %d/%m/%Y"),
+                                        end_date=end_date.strftime("%H:%M %d/%m/%Y")
+                                    )
```
Comment (stefan-walluhn) [inline radio/apps/schedules/models.py]: datetime.strftime is already dedicated to perform date string formating. We do not need to increase complexity to format the date string a second time on a multiple levels.

File: radio/apps/schedules/models.py [hunk 14]
```diff
@@ -230,23 +220,22 @@ def save(self, *args, **kwargs):
 
     @classmethod
     def between(cls, after, before, exclude=None, live=False, schedule_board=None):
-        list_schedules = cls.objects.filter(
-            programme__start_date__lte=before, programme__end_date__isnull=True
-        ) | cls.objects.filter(
-            programme__start_date__lte=before, programme__end_date__gte=after
+        list_schedules = (
+            cls.objects.filter(programme__start_date__lte=before, programme__end_date__isnull=True) |
+            cls.objects.filter(programme__start_date__lte=before, programme__end_date__gte=after)
```
Comment (stefan-walluhn) [inline radio/apps/schedules/models.py]: Make code more readable. The important part is to apply multiple filters, so the code should not spread this across multiple indentation levels, but should reflect this by intuitive reading.

File: radio/apps/schedules/models.py [hunk 15]
```diff
@@ -230,23 +220,22 @@ def save(self, *args, **kwargs):
...
         if schedule_board:
             list_schedules = list_schedules.filter(schedule_board=schedule_board)
         else:
-            list_schedules = list_schedules.filter(
-                schedule_board__start_date__lte=before, schedule_board__end_date__isnull=True
-            ) | list_schedules.filter(
-                schedule_board__start_date__lte=before, schedule_board__end_date__gte=after
+            list_schedules = (
+                list_schedules.filter(schedule_board__start_date__lte=before, schedule_board__end_date__isnull=True) |
+                list_schedules.filter(schedule_board__start_date__lte=before, schedule_board__end_date__gte=after)
```
Comment (stefan-walluhn) [inline radio/apps/schedules/models.py]: Make code more readable. The important part is to apply multiple filters, so the code should not spread this across multiple indentation levels, but should reflect this by intuitive reading.

File: radio/apps/schedules/models.py [hunk 16]
```diff
@@ -262,10 +251,9 @@ def between(cls, after, before, exclude=None, live=False, schedule_board=None):
 
     @classmethod
     def schedule(cls, dt, exclude=None):
-        list_schedules = cls.objects.filter(
-            programme__start_date__lte=dt, programme__end_date__isnull=True
-        ) | cls.objects.filter(
-            programme__start_date__lte=dt, programme__end_date__gte=dt
+        list_schedules = (
+            cls.objects.filter(programme__start_date__lte=dt, programme__end_date__isnull=True) |
+            cls.objects.filter(programme__start_date__lte=dt, programme__end_date__gte=dt)
```
Comment (stefan-walluhn) [inline radio/apps/schedules/models.py]: Make code more readable. The important part is to apply multiple filters, so the code should not spread this across multiple indentation levels, but should reflect this by intuitive reading.

File: radio/apps/schedules/models.py [hunk 17]
```diff
@@ -285,20 +272,17 @@ def schedule(cls, dt, exclude=None):
     @classmethod
     def get_next_date(cls, programme, after):
         list_schedules = cls.objects.filter(programme=programme, type='L')
-        list_schedules = list_schedules.filter(
-            programme__end_date__isnull=True
-        ) | list_schedules.filter(
-            programme__end_date__gte=after
+        list_schedules = (
+            list_schedules.filter(programme__end_date__isnull=True) |
+            list_schedules.filter(programme__end_date__gte=after)
```
Comment (stefan-walluhn) [inline radio/apps/schedules/models.py]: Make code more readable. The important part is to apply multiple filters, so the code should not spread this across multiple indentation levels, but should reflect this by intuitive reading.

File: radio/apps/schedules/models.py [hunk 18]
```diff
@@ -285,20 +272,17 @@ def schedule(cls, dt, exclude=None):
...
         )
-        list_schedules = list_schedules.filter(
-            schedule_board__start_date__isnull=False, schedule_board__end_date__isnull=True
-        ) | list_schedules.filter(
-            schedule_board__end_date__gte=after
+        list_schedules = (
+            list_schedules.filter(schedule_board__start_date__isnull=False, schedule_board__end_date__isnull=True) |
+            list_schedules.filter(schedule_board__end_date__gte=after)
```
Comment (stefan-walluhn) [inline radio/apps/schedules/models.py]: Make code more readable. The important part is to apply multiple filters, so the code should not spread this across multiple indentation levels, but should reflect this by intuitive reading.

File: radio/apps/schedules/models.py [hunk 19]
```diff
@@ -308,7 +292,7 @@ def get_next_date(cls, programme, after):
         return closer_schedule, closer_date
 
     def __unicode__(self):
-        return self.get_day_display() + ' - ' + self.start_hour.strftime('%H:%M')
+        return ' - '.join([self.get_day_display(), self.start_hour.strftime('%H:%M')])
```
Comment (stefan-walluhn) [inline radio/apps/schedules/models.py]: This is classical string joining, so no need for hacky string arithmetic.

File: radio/apps/schedules/tests.py [hunk 20]
```diff
@@ -21,8 +21,9 @@
 from django.core.exceptions import ValidationError
 from django.test import TestCase
 
-from apps.programmes.models import Programme
-from apps.schedules.models import ScheduleBoard, Schedule, MO, TU, WE, TH, FR, SA, SU
+from apps.programmes.models import Programme, Episode
+from apps.schedules.models import ScheduleBoard, Schedule
+from apps.schedules.models import MO, TU, WE, TH, FR, SA, SU
```
Comment (stefan-walluhn) [inline radio/apps/schedules/tests.py]: Extract weekday import to underline these are different things to import (classes vs. constants)

File: radio/apps/schedules/tests.py [hunk 21]
```diff
@@ -43,28 +44,12 @@ def setUp(self):
...
-        Schedule.objects.create(
-            programme=midnight_programme, day=SA, start_hour=start_hour, type='L', schedule_board=schedule_board
-        )
-        Schedule.objects.create(
-            programme=midnight_programme, day=SU, start_hour=start_hour, type='L', schedule_board=schedule_board
-        )
+        for day in (MO, TU, WE, TH, FR, SA, SU):
+            Schedule.objects.create(
+                programme=midnight_programme, day=day, start_hour=start_hour, type='L', schedule_board=schedule_board
+            )
```
Comment (stefan-walluhn) [inline radio/apps/schedules/tests.py]: DRY. Repeating code -> Loop

File: radio/apps/schedules/tests.py [hunk 22]
```diff
@@ -72,15 +57,12 @@ def setUp(self):
...
-        )
-        Schedule.objects.create(
-            programme=programme, day=FR, start_hour=datetime.time(9, 0, 0), type='L', schedule_board=schedule_board
-        )
+
+        for day in (MO, WE, FR):
+            Schedule.objects.create(
+                programme=programme, day=day, start_hour=datetime.time(9, 0, 0), type='L',
+                schedule_board=schedule_board
+            )
```
Comment (stefan-walluhn) [inline radio/apps/schedules/tests.py]: DRY. Repeating code -> loop

File: radio/apps/schedules/tests.py [hunk 23]
```diff
@@ -88,15 +70,12 @@ def setUp(self):
...
-        )
-        Schedule.objects.create(
-            programme=programme, day=FR, start_hour=datetime.time(10, 0, 0), type='L', schedule_board=schedule_board
-        )
+
+        for day in (MO, WE, FR):
+            Schedule.objects.create(
+                programme=programme, day=day, start_hour=datetime.time(10, 0, 0), type='L',
+                schedule_board=schedule_board
+            )
```
Comment (stefan-walluhn) [inline radio/apps/schedules/tests.py]: DRY. Repeating code -> loop

File: radio/apps/schedules/urls.py [hunk 24]
```diff
@@ -19,7 +19,7 @@
 
 from apps.schedules import views
 
-urlpatterns = patterns('',
+urlpatterns = [
     url(r'^$', views.schedule_list, name='list'),
     url(r'^feed_schedules/', views.feed_schedules, name='feed_schedules'),
-)
+]
```
Comment (stefan-walluhn) [inline radio/apps/schedules/urls.py]: patterns() is deprecated (https://docs.djangoproject.com/en/1.9/ref/urls/#patterns)

File: radio/apps/schedules/views.py [hunk 25]
```diff
@@ -49,11 +49,10 @@ def feed_schedules(request):
...
 
 
-background_colours = {"L": "#F9AD81", "B": "#C4DF9B", "S": "#8493CA"}
-text_colours = {"L": "black", "B": "black", "S": "black"}
-
-
 def __get_events(after, before, json_mode=False):
+    background_colours = {"L": "#F9AD81", "B": "#C4DF9B", "S": "#8493CA"}
+    text_colours = {"L": "black", "B": "black", "S": "black"}
+
```
Comment (stefan-walluhn) [inline radio/apps/schedules/views.py]: background_colours and text_colours are local variables, no need to poison global variable space.

**Comment by iago1460** [comment]
Thanks, looks good, sorry for the extra work. 

I revert the changes on the initial migration file because I don't see the point of modifying that file but let me know If you disagree.

Cheers

**Comment by stefan-walluhn** [comment]
Hi Iago,

thanks a lot for merging. This helps me a lot to work on a clean code path again. Looking onto a great collaboration. I think it's quite normal that distant co-working needs some time to find smooth processes, so let's stay in contact for further development.

Best regards.

**Pull request merged**